test/e2e/packagemanifest_e2e_test.go

```diff
@@ -3,170 +3,181 @@ package e2e
+			Expect(pm.GetLabels()["operatorframework.io/os.linux"]).Should(Equal("supported"))
+
+			// Get a PackageManifestList and ensure it has the correct items
+			pmList, err := pmc.OperatorsV1().PackageManifests(testNamespace).List(context.TODO(), metav1.ListOptions{})
+			Expect(err).NotTo(HaveOccurred(), "could not access package manifests list meta")
+			Expect(pmList.ListMeta).ShouldNot(BeNil(), "package manifest list metadata empty")
+			Expect(pmList.Items).ShouldNot(BeNil())
+		})
+
+	})
 
-		source, err := crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Create(context.TODO(), source, metav1.CreateOptions{})
-		require.NoError(GinkgoT(), err)
-		defer func() {
-			require.NoError(GinkgoT(), crc.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Delete(context.TODO(), source.GetName(), metav1.DeleteOptions{}))
-		}()
-
-		// Wait for package-server to be ready
-		err = wait.Poll(pollInterval, 1*time.Minute, func() (bool, error) {
-			GinkgoT().Logf("Polling package-server...")
-			_, err := pmc.OperatorsV1().PackageManifests(testNamespace).List(context.TODO(), metav1.ListOptions{})
-			if err == nil {
-				return true, nil
+	Context("Given a CatalogSource created using gRPC catalog source type", func() {
+		var (
+			packageName   string
+			catalogSource *v1alpha1.CatalogSource
+		)
+		BeforeEach(func() {
+			sourceName := genName("catalog-")
+			packageName = "etcd-test"
+			image := "quay.io/olmtest/catsrc-update-test:related"
+
+			catalogSource = &v1alpha1.CatalogSource{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       v1alpha1.CatalogSourceKind,
+					APIVersion: v1alpha1.CatalogSourceCRDAPIVersion,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      sourceName,
+					Namespace: testNamespace,
+					Labels:    map[string]string{"olm.catalogSource": sourceName},
+				},
+				Spec: v1alpha1.CatalogSourceSpec{
+					SourceType: v1alpha1.SourceTypeGrpc,
+					Image:      image,
+				},
 			}
-			return false, nil
+
+			var err error
+			catalogSource, err = crc.OperatorsV1alpha1().CatalogSources(catalogSource.GetNamespace()).Create(context.TODO(), catalogSource, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
 		})
-		require.NoError(GinkgoT(), err, "package-server not available")
 
-		_, err = fetchCatalogSource(GinkgoT(), crc, source.GetName(), testNamespace, catalogSourceRegistryPodSynced)
-		require.NoError(GinkgoT(), err)
+		AfterEach(func() {
+			err := crc.OperatorsV1alpha1().CatalogSources(catalogSource.GetNamespace()).Delete(context.TODO(), catalogSource.GetName(), metav1.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		})
 
-		pm, err := fetchPackageManifest(GinkgoT(), pmc, testNamespace, packageName, packageManifestHasStatus)
-		require.NoError(GinkgoT(), err, "error getting package manifest")
-		require.NotNil(GinkgoT(), pm)
-		require.Equal(GinkgoT(), packageName, pm.GetName())
+		It("lists the CatalogSource contents using the PackageManifest API", func() {
```

Comment:
Is this precisely stating the behavior the test is intended to verify? It looks like a `relatedImages` test.